File: CPP/program/Shirt.cpp
//import library
#include <iostream>
#include <string>
#include "Clothing.cpp"

//using standar namespace
using namespace std;

class Shirt : public Clothing{ //class Shirt

    //Private attributes
    private:
        string color; 
        string sleeve_type;
        
    //public methods
    public:
        Shirt(){
            this->color = "";
            this->sleeve_type = "";
        }

        Shirt(string color, string sleeve_type){
            this->color = color;
            this->sleeve_type = sleeve_type;
        }

        string get_color() { return color; }
        string get_sleeve_type() { return sleeve_type; }

        void set_color(string color) {
        this->color = color; 
        }
        void set_sleeve_type(string sleeve_type) {
        this->sleeve_type = sleeve_type; 
        }

        //destractor
        ~Shirt(){
            
        }
};

File: PHP/program/Shirt.php
<!-- Kelas Shirt -->
<?php

require('Clothing.php');    //parent

class Shirt extends Clothing{
    private $color = '';
    private $sleeve_type = '';

    // Constructor
    public function __construct($idProduct, $name, $brand, $price, $size, $material, $gender, $color, $sleeve_type) {
        parent::__construct($idProduct, $name, $brand, $price, $size, $material, $gender);
        $this->color = $color;
        $this->sleeve_type = $sleeve_type;
    }

    // Setter untuk color
    public function setColor($color) {
        $this->color = $color;
    }

    // Getter untuk color
    public function getColor() {
        return $this->color;
    }

    // Setter untuk sleeve_type
    public function setSleeveType($sleeve_type) {
        $this->sleeve_type = $sleeve_type;
    }

    // Getter untuk sleeve_type
    public function getSleeveType() {
        return $this->sleeve_type;
    }
}
?>
